test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -195,61 +197,59 @@ public void testMainPipeline() throws Exception {
         String clustObj3Name = "clusters3";
         ExpressionMatrix data = new ExpressionMatrix().withType("log-ratio").withScale("1.0")
                 .withData(getSampleMatrix());
-        wscl.saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
+        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
                 new ObjectSaveData().withName(exprObjName).withType("KBaseFeatureValues.ExpressionMatrix")
-                .withData(new UObject(data)))));
-        //Assert.assertNotNull("exprObjName ",testWsName + "/" + exprObjName);
-        //Assert.assertEquals("exprObjName ",testWsName + "/" + exprObjName);
-        //Assert.assertNotNull("estimObjName ", estimObjName);
-        //
+                        .withData(new UObject(data)))));
         /////////////// estimate K /////////////////
         EstimateKParams ekp = new EstimateKParams().withInputMatrix(testWsName + "/" +
                 exprObjName).withOutWorkspace(testWsName).withOutEstimateResult(estimObjName);
-        impl.estimateK(ekp, token, getContext());
-        ObjectData res1 = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
```

Comment:
Why are these lines deleted? I assume these are checking that the object saved by the method is correct? That's still necessary, right?

Reply:
I see there's many cases of this, I won't point them all out.

Reply:
this is deleted because it is now superseded by the test based on the object from the impl method return

Reply:
I don't understand this. AIUI, the app saves the matrix to the workspace. The changes here make the app also return the matrix at the end of the function. Testing the contents of the matrix returned by the function does not test that the matrix was saved to the ws correctly, or even at all.

Or do I not understand what the function is doing?

Reply:
The change is that it returns an EstimateKResult. What is being tested are the fields in this returned object, for both of the estimateK methods. No matrices here BTW ...

Reply:
> The change is that it returns an EstimateKResult. 

I understand that, but it's still saving the EstimateKResult, right? So why are the tests for that deleted?

Reply:
because testing the saved vs the returned object seems redundant per discussions above?

Reply:
ok, well the object created in the code vs saved in the workspace does make sense. I replicated those tests for the two estimateK methods in the new PR.